DDCore/src/DetectorImp.cpp

```diff
@@ -571,15 +572,15 @@ vector<string> DetectorImp::detectorTypes() const  {
 }
 
 /// Access a set of subdetectors according to the sensitive type.
-const vector<DetElement>& DetectorImp::detectors(const string& type, bool throw_exc)  {
+const vector<DetElement>& DetectorImp::detectors(const string& type, bool throw_exc) const {
   if ( m_manager->IsClosed() ) {
     if ( throw_exc )  {
       DetectorTypeMap::const_iterator i=m_detectorTypes.find(type);
       if ( i != m_detectorTypes.end() ) return (*i).second;
       throw runtime_error("detectors("+type+"): Detectors of this type do not exist in the current setup!");
     }
     // return empty vector instead of exception
-    return m_detectorTypes[ type ] ;
+    return m_detectorTypes.at("") ;
```

Comment:
Now if throw_exc is false empty vector is always returned.
Rather
```c++
  static vector<DetElement> empty;
  if ( m_manager->IsClosed() ) {
    DetectorTypeMap::const_iterator i=m_detectorTypes.find(type);
    if ( i != m_detectorTypes.end() ) return (*i).second;
    if (throw_exc) {
      throw runtime_error("detectors("+type+"): Detectors of this type do not exist in   the current setup!");
     }
    // return empty vector instead of exception
    return empty;
  }
```

Reply:
Thanks, I guess my mind had that logic but the code didn't ;-)